modules/collect/src/main/java/com/opengamma/strata/collect/io/CsvOutput.java

```diff
@@ -132,17 +133,28 @@ private String formatEntry(String entry, boolean alwaysQuote) {
     }
   }
 
-  // quoting is required if entry contains quote, comma or trimmable whitespace
-  private boolean isQuotingRequired(String line) {
-    return line.indexOf('"') >= 0 || line.indexOf(',') >= 0 || line.trim().length() != line.length();
+  // quoting is required if entry contains quote, comma, trimmable whitespace, or starts with an expression character
+  private boolean isQuotingRequired(String entry) {
+    return entry.indexOf('"') >= 0 ||
+        entry.indexOf(',') >= 0 ||
+        entry.trim().length() != entry.length() ||
+        isExpressionPrefix(entry);
+  }
+
+  private boolean isExpressionPrefix(String entry) {
+    char first = entry.length() == 0 ? ' ' : entry.charAt(0);
```

Comment:
Feels odd to be assigning empty char here given we know empty char will always return false in the subsequent check. Admittedly the alternatives would require more code but this is abit random IMO.